packages/react-scripts/template/README.md

```diff
@@ -850,6 +851,76 @@ In the future we plan to integrate it into Create React App even more closely.
 
 To learn more about Flow, check out [its documentation](https://flow.org/).
 
+## Adding TypeScript
+
+TypeScript is a typed superset of JavaScript that compiles to plain JavaScript.
+
+Recent versions of [TypeScript](https://www.typescriptlang.org/) work with Create React App projects out of the box thanks to Babel 7. Beware that Babel 7 TypeScript does not allow some features of TypeScript such as constant enum and namespaces.
+
+To add TypeScript to a Create React App project, follow these steps:
+
+1. Run `npm install --dev typescript @types/react @types/react-dom @types/jest` (or `yarn add --dev typescript @types/react @types/react-dom @types/jest`).
+2. Rename the `.js` files you want to convert. Use `.tsx` if they use JSX or `.ts` if not (e.g. `git mv src/index.js src/index.tsx`).
+
+3. Create a `tsconfig.json` file at the root directory with the following content:
+
+```json
+{
+  "compilerOptions": {
+    "target": "esnext",
+    "module": "esnext",
+    "moduleResolution": "node",
+    "lib": ["esnext", "dom"],
```

Comment:
This assumes the user is polyfilling everything, and is also missing `dom.iterable`

Reply:
> also missing dom.iterable

It has `"dom"`, isn't it included?

Reply:
```ts
new Set(document.querySelectorAll('meta'))
```

for example. `NodeList`s are iterable but that might not be included in the types if `dom.iterable` is not also loaded.